frontend-react/src/components/Admin/CompareJsonModal.test.tsx

```diff
@@ -0,0 +1,237 @@
+                renderComponent();
+
+                fireEvent.change(textareaNode, {
+                    target: { value: INVALID_JSON },
+                });
+            });
+
+            test("disables the save button", () => {
+                expect(saveButtonNode).toBeDisabled();
+            });
+        });
+    });
+
+    describe("on blur", () => {
+        describe("when the updated JSON is valid", () => {
+            beforeEach(() => {
+                renderComponent();
+
+                fireEvent.change(textareaNode, {
+                    target: { value: VALID_JSON_UPDATED },
+                });
+                fireEvent.blur(textareaNode);
+
+                errorDiffNode = screen.queryByTestId(
+                    "EditableCompare__errorDiff"
+                );
+            });
+
+            test("does not render an error diff", () => {
+                expect(errorDiffNode).toBeNull();
+            });
+
+            test("does not render an error toast", () => {
+                expect(
+                    screen.queryByText(/JSON data generated/)
+                ).not.toBeInTheDocument();
+            });
+        });
+
+        describe("when the updated JSON is invalid", () => {
+            const consoleTraceSpy = jest.fn();
+
+            beforeEach(() => {
+                renderComponent();
+
+                jest.spyOn(console, "trace").mockImplementationOnce(
+                    consoleTraceSpy
+                );
+
+                fireEvent.change(textareaNode, {
+                    target: { value: INVALID_JSON },
+                });
+                fireEvent.blur(textareaNode);
+
+                errorDiffNode = screen.getByTestId(
+                    "EditableCompare__errorDiff"
+                );
+            });
+
+            afterEach(() => {
+                jest.resetAllMocks();
+            });
+
+            test("renders an error diff highlighting the error", () => {
+                expect(errorDiffNode).toBeVisible();
+                expect(errorDiffNode?.innerHTML).toContain("{ nope");
+            });
+
+            test("renders an error toast", () => {
+                expect(consoleTraceSpy).toHaveBeenCalled();
+                expect(
+                    screen.queryByText(/JSON data generated/)
+                ).not.toBeInTheDocument();
+            });
+
+            describe("when the user starts typing again", () => {
+                test("it removes the highlighting", () => {
+                    expect(errorDiffNode).toBeVisible();
+
+                    fireEvent.change(textareaNode, {
+                        target: { value: "lalalalala" },
+                    });
+
+                    errorDiffNode = screen.queryByTestId(
+                        "EditableCompare__errorDiff"
+                    );
+                    expect(errorDiffNode).toBeNull();
+                });
+            });
+
+            test("when the user starts typing again", () => {});
```

Comment:
Was this left here as a placeholder?

Reply:
Whoops, nope!  That's just my eyes failing me :) Thanks for catching -- will remove!

Reply:
Removed in this commit: https://github.com/CDCgov/prime-reportstream/pull/7713/commits/f58b30724999df81bf11614f398e907531f53e8e